src/app.md

```diff
@@ -725,6 +726,55 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-latest]: #ex-semver-latest
+<a name="ex-semver-latest"></a>
+## Find the latest version satisfying given range
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Given a list of version `&str`s, find the latest [`semver::Version`] that satisfies the given range
+"<= 1.0.0". A [`semver::VersionReq`] is used to compare the parsed version to a 
+minimum version requirement. Filter for version `&str` that are verified as [`semver::Version`] with [`Version::parse`] and satisfies the [`semver::VersionReq`] using [`VersionReq::matches`]. Return the latest [`semver::Version`] using [`max`].
```

Comment:
I'd probably refrain from describing all of the code step by step (especially the last sentence might be omitted).

Reply:
@budziq @ludwigpacifici I updated the description so that it is concise.

Regarding an example with the pre-release tags, I am working on that in a separate branch and will submit as a PR when it is working.